services/frontend-service/src/ui/components/EnvironmentConfigDialog/EnvironmentConfigDialog.test.tsx

```diff
@@ -0,0 +1,128 @@
+    environmentConfigDialogConfigClass,
+} from './EnvironmentConfigDialog';
+
+jest.mock('../../../api/api').mock('../../utils/GrpcApi');
+
+const configContent = 'config content';
+const configLoadError = 'fake loading error'; //new Error('error loading config');
+const mockGetEnvironmentConfigPretty = jest.fn();
+const mockShowSnackbarError = jest.fn();
+jest.mock('../../utils/store', () => ({
+    GetEnvironmentConfigPretty: () => mockGetEnvironmentConfigPretty(),
+    showSnackbarError: () => mockShowSnackbarError(),
+}));
+
+const openEnvironmentDialog = 'open environment';
+const closedEnvironmentDialog = 'closed environment';
+jest.mock('../../utils/Links', () => ({
+    getOpenEnvironmentConfigDialog() {
+        return openEnvironmentDialog;
+    },
+}));
+
+describe('EnvironmentConfigDialog', () => {
+    const getNode = (environmentName: string) => (
+        <MemoryRouter>
+            <EnvironmentConfigDialog environmentName={environmentName} />
+        </MemoryRouter>
+    );
+
+    type TestData = {
+        name: string;
+        environmentName: string;
+        config: () => Promise<string>;
+        expectedNumDialogs: number;
+        expectedNumConfigs: number;
+        expectedConfig: string;
+        expectedNumSpinners: number;
+        expectedSnackbarErrorCalls: number;
+    };
+
+    const data: TestData[] = [
+        {
+            name: 'closed environment config',
+            environmentName: closedEnvironmentDialog,
+            config: () => Promise.resolve(configContent),
+            expectedNumDialogs: 0,
+            expectedNumConfigs: 0,
+            expectedConfig: configContent,
+            expectedNumSpinners: 0,
+            expectedSnackbarErrorCalls: 0,
+        },
+        {
+            name: 'open environment config',
+            environmentName: openEnvironmentDialog,
+            config: () => Promise.resolve(configContent),
+            expectedNumDialogs: 1,
+            expectedNumConfigs: 1,
+            expectedConfig: configContent,
+            expectedNumSpinners: 0,
+            expectedSnackbarErrorCalls: 0,
+        },
+        {
+            name: 'loading environment config',
+            environmentName: openEnvironmentDialog,
+            config: () => new Promise(() => {}), // never resolves
+            expectedNumDialogs: 1,
+            expectedNumConfigs: 0,
+            expectedConfig: configContent,
+            expectedNumSpinners: 1,
+            expectedSnackbarErrorCalls: 0,
+        },
+        {
+            name: 'failed loading environment config',
+            environmentName: openEnvironmentDialog,
+            config: () => Promise.reject(configLoadError),
+            expectedNumDialogs: 1,
+            expectedNumConfigs: 1,
+            expectedConfig: '',
+            expectedNumSpinners: 0,
+            expectedSnackbarErrorCalls: 1,
+        },
+    ];
+
+    describe.each(data)(`Renders an environment config dialog`, (testcase) => {
```

Comment:
don't nest "describe"s why not just using `it.each`?

https://revolution.dev/app/-JqFGExX46gs9mH7vxR5/WORKSPACE_DOCUMENT/-NR3VEwb1rTCzQhI3h2H/